crypto/mem_sec.c

```diff
@@ -57,10 +56,6 @@ int CRYPTO_secure_malloc_init(size_t size, int minsize)
 #ifdef IMPLEMENTED
     int ret = 0;
 
-    if (too_late)
-        return ret;
```

Comment:
The too_late variables used to prevent freeing incorrectly; but the correct free method can be determined by checking to see if the pointer is in the arena. A problem that has not been solved is when secure memory is disabled, but not everything has been freed. Trying to free then would fail, but I don't believe that behavior has changed. To be safer, the CRYPTO_secure_malloc_done() call should fail if there is still secure memory allocated.